src/rabbit_registry.erl

```diff
@@ -114,6 +116,29 @@ conditional_unregister({exchange_decorator, Type}) ->
 conditional_unregister(_) ->
     ok.
 
+% update channels to enable/disable interceptors
+registered({{Class, Type}, ModuleName}) ->
+    ClassModule = class_module(Class),
+    case proplists:get_value(registered, ClassModule:module_info(functions)) of
```

Comment:
Behaviours tend to use short names but function name conflicts in multi-behaviour modules are also a known problem. If we were to use a longer name, I'd go with `added_to_rabbit_registry` and such. At least this way we are as specific as we can :)